### ISSUE TITLE
[PAID] [$1000] Setting - Console error appears and avatar is broken when uploading a photo to the profile

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#16354]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Access staging.new.expensify.com
2. Sign into a valid account
3. Open your JS console
4. Go to Settings > Profile > Click on profile photo
5. Click on Remove photo
6. Click on the profile photo again > Click on Upload photo
7. Select photo and see JS console errors,
8. Close profile window and see profile photo isn't updated at the top left,

**Image:**

![Sombrero cat (1)](https://user-images.githubusercontent.com/93399543/226724864-b156771f-c7ce-4ef9-8e27-3735d025d847.png)

## Expected Result:
The profile photo should be display and there should be no error in the console

## Actual Result:
Console error appears and avatar is broken when uploading a photo to the profile

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.88.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** NO

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/226725591-781098ee-d0bf-4feb-87e0-2d94ce8329e5.mp4

![image (25)](https://user-images.githubusercontent.com/93399543/226725655-be9443f2-2b61-49b2-a250-bba21affed0c.png)

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014508d42c3166d105</li>
        <li>Upwork Job ID: 1638299901026033664</li>
        <li>Last Price Increase: 2023-03-21</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~014508d42c3166d105

**Comment by MelvinBot** [comment]
Current assignee @strepanier03 is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @s77rt (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @yuwenmemon (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by dayyad** [comment]
## Proposal

### Problem: Users are unable to successfully upload a profile photo in the React project, resulting in JavaScript console errors and a failure to update the displayed profile photo.

### Root Cause: The root cause of this issue might be related to CORS configuration, improper handling of Blob URLs, or incorrect state updates in the application.

### Proposed Solution:

    CORS issues: Verify if the server is configured correctly to add the necessary CORS headers, allowing cross-origin access. Implement CORS middleware in the server-side code if needed.

    Blob URL handling: Ensure proper creation and revoking of Blob URLs by utilizing the URL.createObjectURL() and URL.revokeObjectURL() methods in the code.

    State updates: Confirm that the state is being correctly updated when the profile photo is uploaded, reflecting the new photo URL and displaying the photo as expected in the application.

### Alternative Solutions Explored: N/A

### Next Steps: Once the proposal is reviewed and approved by the Expensify team, I will proceed to implement the changes and create a pull request for review.

**Comment by MelvinBot** [comment]
📣 @dayyad! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by dayyad** [comment]
Contributor details
Your Expensify account email: jonahease@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/jonahe

**Comment by dayyad** [comment]
Contributor details
Your Expensify account email: [jonahease@gmail.com](mailto:jonahease@gmail.com)
Upwork Profile Link: https://www.upwork.com/freelancers/~01e2b4d8accc1f40d3

**Comment by MelvinBot** [comment]
⚠️ Missing/invalid email or upwork profile link. Please make sure you add both your Expensify email and Upwork profile link in the format specified.

**Comment by dayyad** [comment]
Contributor details
Your Expensify account email: jonahease@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01e2b4d8accc1f40d3

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by hellohublot** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Setting - Console error appears and avatar is broken when uploading a photo to the profile

### What is the root cause of that problem?

https://github.com/Expensify/App/pull/15680

We add the picture size param to the url suffix, let `cloudfront.net` help us process and return the picture of the right size, and display it on the small avatar of LHN. If the picture starts with 'https://xxxx.cloudfront.net/xxx.png', it does work well

But when image is uploading, `onyxDB.personalDetails.avatar` temporarily saves with the url `blob:https://staging.new.expensify.com/XXXX-XXXX`, so add `_128` in url will cause problems



### What changes do you think we should make in order to solve the problem?

In fact, only cloudfront.net can receive such a `_128` suffix and handle the image size, so we judge whether this image server is used

https://github.com/Expensify/App/blob/ebc03dbb14d1689e35301278833a82716ae53034/src/libs/ReportUtils.js#L573

``` diff
+  const imageServerDomainMatch = '^https://\w+cloudfront.net'
+  if (!new RegExp(imageServerDomainMatch).test(avatarURL)) {
+       return source;
+  }
```

Other cases to consider
- In the case of offline network upload, `onyxDB.personalDetails.avatar` is always `blob://` because no push message is received, so there is no way to always scale this small size
- If we received `file:///var/xxx.png` on the native mobile platform
- There are some `_128` in the `ReportUtils` code, maybe we can refactor to convertImgeUrl(url, size)


### What alternative solutions did you explore? (Optional)
Not Yet

**Comment by kbecciv** [comment]
@strepanier03 This is issue ONLY in stsging, did not use Deploy broker due to PR related.

**Comment by s77rt** [comment]
@dayyad Thanks for the proposal. I think your RCA is invalid. You need to pinpoint the exact root cause and not list random potential causes.

**Comment by s77rt** [comment]
@fedirjh Thanks for the proposal. Your RCA is correct however the fix is not solid enough as we may encounter the same issue with other links besides those that start with `blob:`.

**Comment by s77rt** [comment]
@hellohublot Thanks for the proposal. Your RCA is correct and the proposed fix looks good to me; I like the idea of relaying on a white list instead of a black list.

:ribbon: :eyes: :ribbon: C+ reviewed

cc @yuwenmemon

**Comment by MelvinBot** [comment]
📣 @hellohublot You have been assigned to this job by @yuwenmemon!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by MelvinBot** [comment]
Triggered auto assignment to @marcochavezf (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@s77rt / @yuwenmemon / @marcochavezf] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@s77rt / @yuwenmemon / @marcochavezf] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@s77rt / @yuwenmemon / @marcochavezf] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@strepanier03] Determine if we should create a regression test for this bug.
- [ ] [@s77rt] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@strepanier03] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: https://github.com/Expensify/App/pull/15680
- The offending PR has been commented on: https://github.com/Expensify/App/pull/15680#issuecomment-1483258859
- A discussion in #expensify-bugs has been started: https://expensify.slack.com/archives/C049HHMV9SM/p1679683646352149

___

**Regression Test Proposal**
1. Open App
2. Go to Settings > Profile > Click on profile photo
3. Click on Remove photo (if a photo is selected already)
4. Click on the profile photo again then Click on Upload photo
5. Select a photo and verify no errors appear in the JS console
6. Close modal and verify profile photo is correctly updated on LHN

**Comment by s77rt** [comment]
@yuwenmemon Please re-open. This is on hold for payment.

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@s77rt / @yuwenmemon / @marcochavezf] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@s77rt / @yuwenmemon / @marcochavezf] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@s77rt / @yuwenmemon / @marcochavezf] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@strepanier03] Determine if we should create a regression test for this bug.
- [ ] [@s77rt] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@strepanier03] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by luacmartins** [comment]
Removing the blocker label since this issue was fixed [here](https://github.com/Expensify/App/pull/16441)

**Comment by s77rt** [comment]
The checklist is has been provided here already https://github.com/Expensify/App/issues/16354#issuecomment-1483264384

**Comment by strepanier03** [comment]
@hellohublot and @s77rt - I have hired you both for the Upwork Job.

----

I've copied over rthe checklist responses into the checklist post, created the reg test GH, and I updated payment amounts ([here](https://github.com/Expensify/App/issues/16354#issuecomment-1484887065)). 

---- 

Will be back to pay this once it's off hold.

**Comment by strepanier03** [comment]
@hellohublot and @s77rt - I have paid out the jobs and ended the contracts. Thank you both for your hard work!
